plugins/providers/metabase/client.go

```diff
@@ -170,6 +172,134 @@ func (c *client) GetCollections() ([]*Collection, error) {
 	return collection, nil
 }
 
+func (c *client) GetGroups() ([]*Group, map[string][]map[string]interface{}, map[string][]map[string]interface{}, error) {
+
+	wg := sync.WaitGroup{}
+	wg.Add(3)
+
+	var groups []*Group
+	var err error
+	go c.fetchGroups(&wg, &groups, err)
+
+	databaseResourceGroups := make(map[string][]map[string]interface{}, 0)
+	go c.fetchDatabasePermissions(&wg, databaseResourceGroups, err)
+
+	collectionResourceGroups := make(map[string][]map[string]interface{}, 0)
+	go c.fetchCollectionPermissions(&wg, collectionResourceGroups, err)
+
+	wg.Wait()
+
+	groupMap := make(map[string]*Group, 0)
+	for _, group := range groups {
+		groupMap[fmt.Sprintf("group:%d", group.ID)] = group
+	}
+
+	addResourceToGroup(databaseResourceGroups, groupMap, "database")
+	addResourceToGroup(collectionResourceGroups, groupMap, "collection")
+
+	return groups, databaseResourceGroups, collectionResourceGroups, err
+}
+
+func (c *client) fetchGroups(wg *sync.WaitGroup, groups *[]*Group, err error) {
```

Comment:
`*[]*Groups`? - can we have a map here itself like the others.

Reply:
just for consistency sake I guess. the other similar ones here use the map to share data... while this one uses slice and then gets converted map ultimately in the caller function anyway.

Reply:
here we are not using map because we want to selected field from /ap/permissions/group API